execution_engine/src/core/runtime/mod.rs

```diff
@@ -2381,6 +2388,13 @@ where
             .context
             .get_validated_contract_package(contract_package_hash)?;
 
+        let version = contract_package.current_contract_version();
+
+        // Return an error if the contract is locked and has some version associated with it.
+        if contract_package.is_locked() && version.is_some() {
```

Comment:
I guess since we want to make contract packages immutable it's also worth to return `LockedContract` err when trying to disable contract's version that's installed onto locked contract package. This way user won't be tempted to disable v1 on locked package, and try to install it again. This way there's won't be chance to modify contract package on global state.

Reply:
Applied lock condition check for `disable_contract_version` as well